cmd/account/iam-cred-rotate.go

```diff
@@ -0,0 +1,1578 @@
+			o.claimName = accountClaim.Spec.AccountLink
+			o.secretName = o.claimName + "-secret"
+			return nil
+		} else {
+			o.log.Warn(o.ctx, "accountClaim.Spec.AccountLink contained empty string, trying resources instead...\n")
+		}
+	}
+	claimName, err = o.getResourcesClaimName()
+	if err != nil {
+		return err
+	}
+	o.claimName = claimName
+	o.secretName = o.claimName + "-secret"
+	return nil
+}
+
+/* fetch aws account claim CR name from resources output
+ * used as a backup to newer 'k8s.GetAccountClaimFromClusterID()'
+ */
+func (o *rotateCredOptions) getResourcesClaimName() (string, error) {
+	o.claimName = ""
+	if o.ocmConn == nil {
+		err := o.preRunSetup()
+		if err != nil {
+			return "", err
+		}
+	}
+	//o.ocmConn.ClustersMgmt().V1().AWSInquiries().STSCredentialRequests().List().Send()
+	liveResponse, err := o.ocmConn.ClustersMgmt().V1().Clusters().Cluster(o.clusterID).Resources().Live().Get().Send()
+	if err != nil {
+		return "", fmt.Errorf("error fetching cluster resources: %w", err)
+	}
+	respBody := liveResponse.Body().Resources()
+	if awsAccountClaim, ok := respBody["aws_account_claim"]; ok {
+		//Unpack into account claim struct...
+		var claimObj awsv1alpha1.AccountClaim
+		err = json.Unmarshal([]byte(awsAccountClaim), &claimObj)
+		if err != nil {
+			o.log.Warn(o.ctx, "Error getting account claim:'%s'", err)
+			return "", err
+		} else {
+			o.log.Debug(o.ctx, "Got awsAccountClaim accountLink:'%s'", claimObj.Spec.AccountLink)
+		}
+		return claimObj.Spec.AccountLink, nil
+	}
+	return "", fmt.Errorf("failed to parse aws_account_claim from api 'clusterMgmt/v1/clusters/%s/resources/live/'", o.clusterID)
+}
+
+func (o *rotateCredOptions) connectClusterClient() error {
+	o.log.Info(o.ctx, "Connect BP client cluster:'%s'\n", o.clusterID)
+	if len(o.reason) <= 0 {
+		return fmt.Errorf("accessing cred secrets requires a reason for elevated command")
+	}
+	// This action requires elevation
+	elevationReasons := []string{}
+	elevationReasons = append(elevationReasons, o.reason)
+	elevationReasons = append(elevationReasons, fmt.Sprintf("Elevation required to rotate secrets '%s' aws-account-cr-name", o.claimName))
+
+	// If some elevationReasons are provided, then the config will be elevated with user backplane-cluster-admin...
+	// Fetch the backplane kubeconfig, ignore the returned kubecli for now...
+	_, kubeConfig, kubClientSet, err := common.GetKubeConfigAndClient(o.clusterID, elevationReasons...)
+	if err != nil {
+		o.log.Warn(o.ctx, "Err creating cluster:'%s' client, GetKubeConfigAndClient() err:'%+v'\n", o.clusterID, err)
+		return err
+	}
+	o.clusterClientset = kubClientSet
+	// Create new kube client with the CredentialsRequest schema using the backplane config from above...
+	scheme := runtime.NewScheme()
+	err = credreqv1.AddToScheme(scheme)
+	if err != nil {
+		o.log.Warn(o.ctx, "Error adding CredentialsRequests to schema:'%s'\n", err)
+		return err
+	}
+	kubeCli, err := client.New(kubeConfig, client.Options{Scheme: scheme})
```

Comment:
Not suggesting you should or need to change it here, but have you tried using `k8s.io/client-go/dynamic` for the client/scheme stuff?  It's pretty sweet. :)

I don't even think it would be an improvement in this particular case.  Just sharing that it's cool, heh.